File: app/src/main/java/com/example/steaminvestmentmanager/utilclasses/DownloadingPageHtmlCode.java
package com.example.steaminvestmentmanager.utilclasses;

import java.net.URL;
import java.net.URLConnection;
import java.util.Scanner;
import java.util.concurrent.Callable;

public class DownloadingPageHtmlCode implements Callable<String> {
    private final String htmlPageUrl;

    public DownloadingPageHtmlCode(String htmlPageUrl) {
        this.htmlPageUrl = htmlPageUrl;
    }

    @Override
    public String call(){
        String content = null;
        URLConnection connection;
        try {
            connection =  new URL(htmlPageUrl).openConnection();
            Scanner scanner = new Scanner(connection.getInputStream());
            scanner.useDelimiter("\\Z");
            content = scanner.next();
            scanner.close();
        }catch ( Exception ignored) {
        }
        return content;
    }
}

File: app/src/main/java/com/example/steaminvestmentmanager/utilclasses/DownloadingPageHtmlCode.kt
package com.example.steaminvestmentmanager.utilclasses

import java.net.URL
import java.net.URLConnection
import java.util.Scanner
import java.util.concurrent.Callable

class DownloadingPageHtmlCode(private val htmlPageUrl: String?) : Callable<String?> {
    override fun call(): String? {
        var content: String? = null
        val connection: URLConnection
        try {
            connection = URL(htmlPageUrl).openConnection()
            val scanner = Scanner(connection.getInputStream())
            scanner.useDelimiter("\\Z")
            content = scanner.next()
            scanner.close()
        } catch (ignored: Exception) {
        }
        return content
    }
}